src/main/java/org/cryptomator/common/mount/MountWithinParentUtil.java

```diff
@@ -22,28 +22,30 @@ public final class MountWithinParentUtil {
 
 	private MountWithinParentUtil() {}
 
-	static void prepareParentNoMountPoint(Path mountPoint) throws MountPointPreparationException {
+	static void prepareParentNoMountPoint(Path mountPoint) throws IllegalMountPointException {
 		Path hideaway = getHideaway(mountPoint);
-		var mpExists = Files.exists(mountPoint, LinkOption.NOFOLLOW_LINKS);
+		var mpExists = removeResidualJunction(mountPoint); //Handle junction as not existing
 		var hideExists = Files.exists(hideaway, LinkOption.NOFOLLOW_LINKS);
 
-		//TODO: possible improvement by just deleting an _empty_ hideaway
-		if (mpExists && hideExists) { //both resources exist (whatever type)
-			throw new MountPointPreparationException(new FileAlreadyExistsException(hideaway.toString()));
-		} else if (!mpExists && !hideExists) { //neither mountpoint nor hideaway exist
-			throw new MountPointPreparationException(new NoSuchFileException(mountPoint.toString()));
+		if (!mpExists && !hideExists) { //neither mountpoint nor hideaway exist
+			throw new MountPointNotExistingException(mountPoint);
 		} else if (!mpExists) { //only hideaway exists
-			checkIsDirectory(hideaway);
+			checkIsHideawayDirectory(mountPoint, hideaway);
 			LOG.info("Mountpoint {} seems to be not properly cleaned up. Will be fixed on unmount.", mountPoint);
 			try {
 				if (SystemUtils.IS_OS_WINDOWS) {
 					Files.setAttribute(hideaway, WIN_HIDDEN_ATTR, true, LinkOption.NOFOLLOW_LINKS);
 				}
 			} catch (IOException e) {
-				throw new MountPointPreparationException(e);
+				throw new UncheckedIOException(e);
```

Comment:
just declare the `IOException` in the method head and don't wrap the IOE

Reply:
Resolved by 821cc0940de3170e4583aeebf17242e6511cfccf.